# DSCI 550 Assignment 1


Katie Chak: chakw@usc.edu

###  Stylometrics

### viii. Attacker language style

2. Random capitalization (test different thresholds)

Check capitalization and period ratio

In [22]:
import pandas as pd
import numpy as np

In [7]:
def read_file(filename):
    with open(filename, 'r') as file:
        data = file.read().replace('\n', '')
    return data

In [12]:
str1=read_file("opinion_1.txt")
str2=read_file("academic_journal_2.txt")
str3=read_file("opinion_2.txt")
str4=read_file("opinion_4.txt")
str5=read_file("opinion_3.txt")
str6=read_file("academic_journal_3.txt")
str7=read_file("diary_4.txt")
str8=read_file("poem_1.txt")
str9=read_file("poem_2.txt")
str10=read_file("poem_3.txt")
str11=read_file("poem_4.txt")

In [18]:
# counts number of capital letters and periods in a string and returns the ratio

def count_period_ratio(string):
    """Input is a string of article"""
    period_cnt=0
    capital_cnt=0
    
    for letter in string:
        if letter==".":
            period_cnt+=1
        elif letter.isupper()==True:
            capital_cnt+=1
    
    ratio=capital_cnt/period_cnt
    return ratio
        

In [14]:
strlist=[str1,str2,str3,str4,str5,str6,str7,str8,str9,str10,str11]

Make a list of ratios from some test strs (articles, academic journals, poems, and diaries)

In [15]:
ratearray=[]
for strs in strlist:
    ratearray.append(count_period_ratio(strs))

In [46]:
df1=pd.DataFrame(ratearray).describe()
df1.columns=["normal_text"]
df1

normal_text
count    11.000000
mean      3.472511
std       1.381239
min       1.903771
25%       2.571014
50%       3.127660
75%       3.761157
max       6.666667

In [58]:
testarray=[]
for i in range (1000):
    try:
        testarray.append(count_period_ratio(messages[i]["X-TIKA:content"].strip().lstrip()))
    except ZeroDivisionError:
        pass
    continue

In [61]:
df2=pd.DataFrame(testarray).describe()
df2.columns=["999_scam_text"]
df2

999_scam_text
count     999.000000
mean       18.358437
std        31.228246
min         0.000000
25%         3.803226
50%         5.448276
75%         9.871711
max       226.142857

In [63]:
df3=pd.concat([df1,df2],axis=1)
df3

normal_text  999_scam_text
count    11.000000     999.000000
mean      3.472511      18.358437
std       1.381239      31.228246
min       1.903771       0.000000
25%       2.571014       3.803226
50%       3.127660       5.448276
75%       3.761157       9.871711
max       6.666667     226.142857

The threshold for irregular capitalization should be >1 (given that one capitalized word would be followed by a period) and <3.76 (75 percentile for regular text)